internal/telemetry/appinsights.go

```diff
@@ -0,0 +1,302 @@
+package telemetry
+
+import (
+	"fmt"
+	"os"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/microsoft/ApplicationInsights-Go/appinsights"
+	"github.com/microsoft/go-sqlcmd/internal/localizer"
+)
+
+var telemetryClient appinsights.TelemetryClient
+var isTelemetryEnabled string = "true"
+
+type TelemetrySqlcmd struct {
+	EventName  string
+	Properties map[string]string
+}
+
+// Event Names
+const (
+	Legacy                  = "sqlcmd/legacy"
+	Config                  = "sqlcmd/config"
+	ConfigAddContext        = "sqlcmd/config/addcontext"
+	ConfigAddEndpoint       = "sqlcmd/config/addendpoint"
+	ConfigAddUser           = "sqlcmd/config/adduser"
+	ConfigConnectionStrings = "sqlcmd/config/connectionstrings"
+	ConfigCurrentContexts   = "sqlcmd/config/currentcontexts"
+	ConfigDeleteContext     = "sqlcmd/config/deletecontext"
+	ConfigDeleteEndpoint    = "sqlcmd/config/deleteendpoint"
+	ConfigDeleteUser        = "sqlcmd/config/deleteuser"
+	ConfigGetContexts       = "sqlcmd/config/getcontexts"
+	ConfigGetEndpoints      = "sqlcmd/config/getendpoints"
+	ConfigGetUsers          = "sqlcmd/config/getusers"
+	ConfigUseContext        = "sqlcmd/config/usecontext"
+	ConfigView              = "sqlcmd/config/view"
+	Create                  = "sqlcmd/create"
+	CreateAzsqlEdge         = "sqlcmd/create/azsqledge"
+	CreateMssql             = "sqlcmd/create/mssql"
+	Delete                  = "sqlcmd/delete"
+	SqlcmdHelp              = "sqlcmd/help"
+	Open                    = "sqlcmd/open"
+	SqlcmdQuery             = "sqlcmd/query"
+	Start                   = "sqlcmd/start"
+	Stop                    = "sqlcmd/stop"
+)
+
+// Property Names
+var (
+	// Common Properties
+	userId    string
+	sessionId string
+	locLang   string
+	userOs    string
+)
+
+const (
+
+	//Common Properties
+	UserId    = "Sqlcmd.userId"
+	SessionId = "Sqlcmd.sessionId"
+	LocLang   = "Sqlcmd.locLang"
+	UserOs    = "Sqlcmd.userOs"
+
+	// Legacy Flags
```

Comment:
>// Legacy Flags [](http://example.com/codeflow?start=1&length=15)

putting these as constants implies maintenance overhead every time we add a new flag.

Is there any reason not to build these identifiers dynamically based on the cobra data? In which scenario will an engineer have to type these constant symbols into a code editor?